packages/app-project/stores/Project.js

```diff
@@ -38,12 +45,16 @@ const Project = types
     },
 
     get defaultWorkflow() {
+      let workflowFromCookie = ''
+      if (typeof document !== 'undefined') {
+        workflowFromCookie = getCookie('workflow_id')
+      }
       const activeWorkflows = self.links['active_workflows']
       let singleActiveWorkflow
       if (activeWorkflows.length === 1) {
         [singleActiveWorkflow] = self.links['active_workflows']
       }
-      return singleActiveWorkflow
+      return singleActiveWorkflow || workflowFromCookie || self.selectedWorkflow
```

Comment:
I'm probably missing something obvious, but `self.selectedWorkflow` here is sometimes empty, after setting the `workflow_id` cookie.

Reply:
Fixed by making sure that `selectedWorkflow` is added to incoming project snapshots.

Reply:
Client-side routing in Next.js applies a new project snapshot to the store from page props on each route change, overwriting any local changes you’ve made to the stored project.